bigtable/docs/snippets_table.py

```diff
@@ -412,6 +412,89 @@ def test_bigtable_table_row():
     table.truncate(timeout=300)
 
 
+def test_bigtable_table_append_row():
+    # [START bigtable_table_append_row]
+    from google.cloud.bigtable import Client
+
+    client = Client(admin=True)
+    instance = client.instance(INSTANCE_ID)
+    table = instance.table(TABLE_ID)
+
+    row_keys = [b"row_key_1", b"row_key_2"]
+    row1_obj = table.append_row(row_keys[0])
+    row2_obj = table.append_row(row_keys[1])
+    # [END bigtable_table_append_row]
+
+    row1_obj.append_cell_value(COLUMN_FAMILY_ID, COL_NAME1, CELL_VAL1)
+    row1_obj.commit()
+    row2_obj.append_cell_value(COLUMN_FAMILY_ID, COL_NAME1, CELL_VAL1)
+    row2_obj.commit()
+
+    actual_rows_keys = []
+    for row in table.read_rows():
+        actual_rows_keys.append(row.row_key)
+
+    assert actual_rows_keys == row_keys
+
+    table.truncate(timeout=300)
+
+
+def test_bigtable_table_direct_row():
+    # [START bigtable_table_direct_row]
+    from google.cloud.bigtable import Client
+
+    client = Client(admin=True)
+    instance = client.instance(INSTANCE_ID)
+    table = instance.table(TABLE_ID)
+
+    row_keys = [b"row_key_1", b"row_key_2"]
+    row1_obj = table.direct_row(row_keys[0])
+    row2_obj = table.direct_row(row_keys[1])
+    # [END bigtable_table_direct_row]
+
+    row1_obj.set_cell(COLUMN_FAMILY_ID, COL_NAME1, CELL_VAL1)
+    row1_obj.commit()
+    row2_obj.set_cell(COLUMN_FAMILY_ID, COL_NAME1, CELL_VAL1)
+    row2_obj.commit()
+
+    actual_rows_keys = []
+    for row in table.read_rows():
+        actual_rows_keys.append(row.row_key)
+
+    assert actual_rows_keys == row_keys
+
+    table.truncate(timeout=300)
+
+
+def test_bigtable_table_conditional_row():
+    # [START bigtable_table_conditional_row]
+    from google.cloud.bigtable import Client
+    from google.cloud.bigtable.row_filters import PassAllFilter
+
+    client = Client(admin=True)
+    instance = client.instance(INSTANCE_ID)
+    table = instance.table(TABLE_ID)
+
+    row_keys = [b"row_key_1", b"row_key_2"]
+    filter_ = PassAllFilter(True)
+    row1_obj = table.conditional_row(row_keys[0], filter_=filter_)
+    row2_obj = table.conditional_row(row_keys[1], filter_=filter_)
+    # [END bigtable_table_conditional_row]
+
+    row1_obj.set_cell(COLUMN_FAMILY_ID, COL_NAME1, CELL_VAL1, state=False)
```

Comment:
@crwilcox I feel like instead of state, we could do something like onMatch=False or onMatch=True

Reply:
@billyjacobson that seems reasonable to me. Does that resemble something we do in other languages? I agree `state` isn't a great name, but it would be good if we didn't diverge from a cross-language pattern if there is one.

Reply:
Node calls it onMatch and onNoMatch.
The RPCs call it true_mutations and false_mutations
Go uses mtrue, mfalse
PHP trueMutations, falseMutations
Ruby true_mutations, false_mutations

So probably true and false mutations would be most consistent

Reply:
We can't just rename the `state` parameter:  it would be a breaking change.

Reply:
Ok